dev/html/public/optimized-appear/defer-handoff-layout-sibling-transform.html

```diff
@@ -0,0 +1,145 @@
+
+        <script type="module">
+            const {
+                motion,
+                animateStyle,
+                animate,
+                startOptimizedAppearAnimation,
+                optimizedAppearDataAttribute,
+                motionValue,
+                frame,
+            } = window.Motion
+            const { matchViewportBox } = window.Assert
+            const root = document.getElementById("root")
+
+            const duration = 1
+            const x = motionValue(0)
+
+            let isFirstFrame = true
+
+            function Component() {
+                const [top, setTop] = React.useState(0)
+                React.useEffect(() => {
+                    setTimeout(() => {
+                        setTop(100)
+                    }, 100)
+                }, [])
+
+                return React.createElement("div", {
+                    id: "container",
+                    children: [
+                        React.createElement(motion.div, {
+                            id: "box",
+                            className: "box",
+                            initial: { x: 0 },
+                            animate: { x: 100 },
+                            transition: { duration, ease: "linear" },
+                            style: { x, position: "relative" },
+                            [optimizedAppearDataAttribute]: "a",
+                            children: "Content",
+                        }),
+                        React.createElement(motion.div, {
+                            id: "layout-box",
+                            className: "box",
+                            initial: { x: 0 },
+                            animate: { x: 100 },
+                            transition: { duration, ease: "linear" },
+                            style: { x, position: "relative", top },
+                            layout: true,
+                            [optimizedAppearDataAttribute]: "b",
+                            onLayoutAnimationStart: () => {
+                                frame.postRender(() => {
+                                    const numAnimationsOnA = document
+                                        .getElementById("box")
+                                        .getAnimations().length
+
+                                    const numAnimationsOnB = document
+                                        .getElementById("layout-box")
+                                        .getAnimations().length
+
+                                    if (
+                                        numAnimationsOnA !== 1 ||
+                                        numAnimationsOnB !== 0
+                                    ) {
+                                        showError(
+                                            document.getElementById("box"),
+                                            "Layout animation on B cancelled optimised animation on A"
```

Comment:
nit: the error can be different, right?
If A !== 1: cancelled by B,
If B !== 0: optimised animation not cancelled?

Reply:
yeah good suggestion